src/twisted/internet/_sslverify.py

```diff
@@ -1481,11 +1580,46 @@ def __init__(self,
         self._mode = SSL.MODE_RELEASE_BUFFERS
 
         if method is None:
-            # If no method is specified set things up so that TLSv1.0 and newer
-            # will be supported.
             self.method = SSL.SSLv23_METHOD
-            self._options |= SSL.OP_NO_SSLv3
+
+            if atLeastTLSVersion:
+                if (reduceSecurityToTLSVersion and
+                    atLeastTLSVersion > reduceSecurityToTLSVersion):
+                    raise ValueError(
+                        ("atLeastTLSVersion needs to be lower than "
+                         "reduceSecurityToTLSVersion"))
+
+                if atLeastTLSVersion > self._defaultMinimumTLSVersion:
+                    noLowerThanTLSVersion = atLeastTLSVersion
+
+            if noLowerThanTLSVersion is None:
+                noLowerThanTLSVersion = self._defaultMinimumTLSVersion
+
+                # If you set the max lower than the default, but don't set the
+                # minimum, pull it down to that
+                if (reduceSecurityToTLSVersion and
+                    noLowerThanTLSVersion > reduceSecurityToTLSVersion):
+                    noLowerThanTLSVersion = reduceSecurityToTLSVersion
+
+            if (reduceSecurityToTLSVersion and
+                noLowerThanTLSVersion > reduceSecurityToTLSVersion):
+                raise ValueError(
+                    ("noLowerThanTLSVersion needs to be lower than "
+                     "reduceSecurityToTLSVersion"))
+
+            excludedVersions = _getExcludedTLSProtocols(
+                noLowerThanTLSVersion, reduceSecurityToTLSVersion)
+
+            for version in excludedVersions:
+                self._options |= _tlsDisableFlags[version]
         else:
+            warnings.warn(
+                ("Passing method to twisted.internet.ssl.CertificateOptions "
+                 "was deprecated in Twisted NEXT. Please use a combination "
+                   "of noLowerThanTLSVersion, atLeastTLSVersion, and "
+                   "reduceSecurityToTLSVersion instead."),
```

Comment:
I think we might want to caution users that they just should stop passing the method entirely and that Twisted will do a good thing by default.  As much as possible, users should accept the defaults :)